src/test/java/seedu/address/logic/commands/DeleteRecurringSessionCommandTest.java

```diff
@@ -105,27 +104,47 @@ void execute_validNameValidIndexValidDate_success() {
 
     @Test
     void equals() {
-        RecurringSession recurringSession =
+        AddressBook addressBook = new AddressBook();
+        AddressBook comparedAddressBook = new AddressBook();
+        Model testingModel = new ModelManager(addressBook, new UserPrefs());
+        Model comparedModel = new ModelManager(comparedAddressBook, new UserPrefs());
+        RecurringSession recurringSessionA =
                 new RecurringSessionBuilder()
                         .withSessionDate(VALID_START_DATE_A, VALID_TIME)
                         .withLastSessionDate(VALID_END_DATE_A, VALID_TIME).build();
-        RecurringSession anotherRecurringSession =
+        RecurringSession copyRecurringSessionA =
+                new RecurringSessionBuilder()
+                        .withSessionDate(VALID_START_DATE_A, VALID_TIME)
+                        .withLastSessionDate(VALID_START_DATE_A, VALID_TIME).build();
+        RecurringSession recurringSessionB =
                 new RecurringSessionBuilder()
                         .withSessionDate(VALID_START_DATE_B, VALID_TIME)
-                        .withLastSessionDate(VALID_END_DATE_B, VALID_TIME).build();
-        Student student = model.getAddressBook().getStudentList().get(INDEX_FIRST_STUDENT.getZeroBased());
-        Student anotherStudent = model.getAddressBook().getStudentList()
-                .get(INDEX_SECOND_STUDENT.getZeroBased());
-        student.addSession(recurringSession);
-        anotherStudent.addSession(anotherRecurringSession);
+                        .withLastSessionDate(VALID_START_DATE_B, VALID_TIME).build();
+        Student exampleStudent1 = new StudentBuilder().withName("Father John").build();
+        Student copyExampleStudent1 = new StudentBuilder().withName("Father John").build();
+        Student exampleStudent2 = new StudentBuilder().withName("Mother Mary").build();
+        testingModel.addStudent(exampleStudent1);
+        comparedModel.addStudent(copyExampleStudent1);
+        testingModel.addStudent(exampleStudent2);
+        exampleStudent1.addSession(recurringSessionA);
+        copyExampleStudent1.addSession(copyRecurringSessionA);
+        exampleStudent2.addSession(recurringSessionA);
+        copyExampleStudent1.addSession(recurringSessionA);
```

Comment:
maybe could group `testingModel`, `comparingModel` stuff together, a lil confusing to follow.

Reply:
Okay, will add more comments and reorder on the next commit!